src/model/RestAction.ts

```diff
@@ -249,7 +277,19 @@ class RestAction implements Action {
                 maxAttempts: 3,
                 retryDelay: 1000, // 1s
                 fullResponse: true,
-            })
+            };
+
+            if (this.clientCertificatePath && this.clientKeyPath) {
```

Comment:
having `getClientCertificationConfiguration` also work on possibly `undefined` paths and simply return an empty object in that case would allow us to make `requestOptions` `const` by directly spreading the result of `getClientCertificationConfiguration` into the initialization of `requestOptions`.